### PR TITLE
feat: obtain final rendering size from parent scene size

### PR DESCRIPTION
**Pull request by yuniruyuni**
# What it is?

- obtaining final rendering size from parent scene item size to implement #2 
- example behavior video:

https://github.com/user-attachments/assets/0221de72-77d3-4200-9c8c-0251564af5bf

# Note

Because of main branch build failure, this implementation contains update of obs-plugin-template to latest.
If you do not want to update it, please read `plugin-main.c` (= `filter-9slice.c` on main branch) change and merge it with single file only. :bow:

### CODE REVIEW

File: src/plugin-main.c [hunk 1]
```diff
@@ -1,342 +1,333 @@
...
+	struct cb_data target = {
+		.parent = parent,
+		.found = NULL,
+	};
+	obs_enum_scenes(enum_scenes_callback, &target);
+
+	struct vec2 scale = {.x = 1.0f, .y = 1.0f};
+	if (target.found) {
+		obs_sceneitem_get_scale(target.found, &scale);
+	}
```
Comment (yuniruyuni) [inline src/plugin-main.c]: Point is this lines. we can obtain scale factor from scene item that searched with parent source name.

**Comment by yuniruyuni** [comment]
(build warning on windows and CI failure (by detecting the warning) was fixed)

**Comment by yuniruyuni** [comment]
if it is too large to review, I'll split CI template update and scaling factor PRs :bow: please take easy to say that 👍

**Comment by nitz** [comment]
Well done figuring this out! Sorry for the delay in reply, I spent way too long fighting with cmake to get it to correctly generate project files. (Turns out it had downloaded a 0 byte zip file that it expected to have things in, but never bothered trying to re-download it until I deleted it because it already existed 🙃)

The CI bit is rather unimportant to me, I just left it since it came with the template 😂

I did just change a few lines that fixed the copyright lines at the top of the sources and added one for you there.

**Comment by yuniruyuni** [comment]
@nitz BTW, could you please create `release` (on github / OBS official) for this plugin update... 🙏  if you want to do this, make `release` on github (or make release tag on git) and then, updated CI will make the final binary for it.

**Pull request merged**